subsys/bootloader/image/CMakeLists.txt

```diff
@@ -29,13 +44,10 @@ endif()
 
 # Include a kconfig file which enables CONFIG_FW_INFO in the image
 # which is booted by B0.
-set(old_conf ${${image_to_boot}OVERLAY_CONFIG})
-string(FIND "${old_conf}" "${CMAKE_CURRENT_SOURCE_DIR}/fw_info.conf" found)
-if (${found} EQUAL -1)
-  set(${image_to_boot}OVERLAY_CONFIG
-    "${old_conf} ${CMAKE_CURRENT_SOURCE_DIR}/fw_info.conf"
-    CACHE STRING "" FORCE)
-endif()
+add_overlay_config(
+  ${image_to_boot}
```

Comment:
Let's either only add the config when image_to_boot is set, or assert that it is set in case we accidentally create a code path in the future where it is unset and we get an obscure error.

Reply:
If "image_to_boot" is not set, then it is the application that is booted. In which case "image_to_boot" correctly is set to "", which make `${image}OVERLAY_CONFIG` in `add_overlay_config()` correctly apply to the application OVERLAY_CONFIG AFAICS.

Reply:
Ah, its too late to affect the kconfig of the application, so you are right, we should assert to inform the user that the fw_info must be set manually in that case.